packages/compartment-mapper/src/import-archive.js

```diff
@@ -97,6 +100,23 @@ const makeArchiveImportHookMaker = (
       // per-module:
       const module = modules[moduleSpecifier];
       if (module === undefined) {
+        if (exitModuleImportHook) {
+          console.error('#################x');
+          const ns = await exitModuleImportHook(moduleSpecifier);
+          if (ns) {
+            return {
+              record: freeze({
+                imports: [],
+                exports: Object.keys(ns),
+                execute: moduleExports => {
+                  moduleExports.default = ns;
+                  Object.assign(moduleExports, ns);
```

Comment:
This uses the `ThirdPartyStaticModuleInterface` style, which might lead to compatibility issues, but until Module work in TC39 gets into the language and SES becomes compatible with it, I don't see a more transparent way to interface any module here regardless of it being imported or required.

We could burden the exitModuleImportHook implementer with returning a record though, in which case they get more freedom to figure out what the default and named exports should be.

I think both versions could work.